app/helpers/form_helper.rb

```diff
@@ -304,16 +304,24 @@ def help_inline(inline, error)
   def add_label options, f, attr
     label_size = options.delete(:label_size) || "col-md-2"
     required_mark = check_required(options, f, attr)
-    label = options[:label] == :none ? '' : options.delete(:label)
-    label ||= ((clazz = f.object.class).respond_to?(:gettext_translation_for_attribute_name) &&
-        s_(clazz.gettext_translation_for_attribute_name attr)) if f
-    label = label.present? ? label_tag(attr, "#{label}#{required_mark}", :class => label_size + " control-label") : ''
+    if options[:label] == :none
+      label = ''.html_safe
+    else
+      label = ''.html_safe + options.delete(:label)
+      label += ((clazz = f.object.class).respond_to?(:gettext_translation_for_attribute_name) &&
+        s_(clazz.gettext_translation_for_attribute_name attr).html_safe) if f && label.empty?
+    end
+
+    if options[:label_help].present?
+      label += ' '.html_safe + popover("", options[:label_help], options[:label_help_options] || {})
+    end
+    label = label.present? ? label_tag(attr, label.to_s + required_mark.to_s, :class => label_size + " control-label") : ''
     label
   end
 
   def check_required options, f, attr
     required = options.delete(:required) # we don't want to use html5 required attr so we delete the option
-    return ' *' if required.nil? ? is_required?(f, attr) : required
+    return ' *'.html_safe if required.nil? ? is_required?(f, attr) : required
```

Comment:
That's true only because this helper is called from `add_label` where we append to safe buffer, I think it's a good practice that helper returns safe strings (if they are safe), at least internal rails helpers do. While we don't do it anywhere in core code, I think this is useful.